client/engine/src/evaluator/mod.rs

```diff
@@ -0,0 +1,899 @@
+use chrono::{DateTime, Utc};
+use snafu::{prelude::*, Whatever};
```

Comment:
@markphelps @yquansah Yeah, I used `anyhow` in that library, because it was kind of a sketch, but long-term I think you'll need to move away from it. Here's what [`anyhow` docs say](https://github.com/dtolnay/anyhow#comparison-to-thiserror) in comparing to `thiserror`:

> Use Anyhow if you don't care what error type your functions return, you just want it to be easy. This is common in application code. Use [thiserror](https://github.com/dtolnay/thiserror) if you are a library that wants to design your own dedicated error type(s) so that on failures the caller gets exactly the information that you choose.

The general guideline at that time (could be an outdated perspective) is to use `anyhow` for applications where users will see errors, but not interrogate them programatically. Libraries might be different, because the error values might be structured information (e.g. nested errors) that needs to be interrogated—in which case use an error crate that supports that attitude.